profiles/killbill/src/main/java/org/killbill/billing/server/log/ThreadNameBasedDiscriminator.java

```diff
@@ -173,9 +200,22 @@ private String getKillbillCaller() {
                     markerName += kar;
                 }
                 return markerName;
+            } else { /* Support for plugins as well */
+                int next = lookupNextToken(classNameChars, 0, "com.killbill.billing.plugin.");
```

Comment:
* How about a system property, for a list of `nextToken`? Custom plugins might use another classname?
* Not sure how Java stores String internally these days: what's the overhead of `toCharArray` (first thing that `lookupNextToken` does)? Should we do it once since the String is constant (this is called a lot, just trying to ensure we don't add too much overhead)?

Reply:
> How about a system property, for a list of nextToken? Custom plugins might use another classname?

There are 2 main difficulties:
1. We would need a rather complex property (a map that says pluginA -> 'com.xxx', ...)
2. While processing this log, the code does not really know which plugin it is coming from - all we do right now is create a stack trace to detect some pattern

> Not sure how Java stores String internally these days: what's the overhead of toCharArray (first thing that lookupNextToken does)? Should we do it once since the String is constant (this is called a lot, just trying to ensure we don't add too much overhead)?

Based on [this comment](https://github.com/killbill/killbill/blob/master/profiles/killbill/src/main/java/org/killbill/billing/server/log/ThreadNameBasedDiscriminator.java#L73) there was a desire speed up things using this scheme - versus doing string comparison. I assumed this was true, and kept the same idea, but I really don't know, whether this extra complexity and code makes sense.

Reply:
Also, one thing I don't really like is that name of the file is relatively random as we pick the package name is a rather large stack trace that fits the patterns we have hardcoded. There is definitely some space for improvements...

Reply:
/cc @pierre  I am merging - but if you want to discuss any of this in more details, please let me know.